modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt

```diff
@@ -243,32 +242,50 @@ open class Iroha2Client(
         }
     }
 
+    fun subscribeToBlockStream(from: Long = 1, count: Int) = subscribeToBlockStream(
+        from,
+        count,
+        action = { block -> block },
+    )
+
     /**
      * Subscribe to block streaming
      * @param from - block number to start from
-     * @param count - how many blocks to get before closing web socket
+     * @param count - how many blocks to get before closing the channel
+     * @param action - code which will be invoked after a new block received
+     * @param closeOn - if the condition returns true then the channel will be closed
      */
-    fun subscribeToBlockStream(from: Long, count: Int): Flow<VersionedBlockMessage> = flow {
+    fun <T> subscribeToBlockStream(
```

Comment:
any comments and suggestions are welcome to make this feature more usable

Reply:
I'd expand that with smth like `onFailure` (optional action on exception)